Wrappers/Python/cil/framework/acquisition_geometry.py

```diff
@@ -1708,22 +1931,40 @@ def dimension(self):
 
     @property
     def shape(self):
+        if self.config.system.geometry != "cone_flex":
+            shape_dict = {AcquisitionDimension.CHANNEL: self.config.channels.num_channels,
+                        AcquisitionDimension.ANGLE: self.config.angles.num_positions,
+                        AcquisitionDimension.VERTICAL: self.config.panel.num_pixels[1],
+                        AcquisitionDimension.HORIZONTAL: self.config.panel.num_pixels[0]}
+        # We are using per-projection geometry, not angles
+        else:
+            shape_dict = {AcquisitionDimension.CHANNEL: self.config.channels.num_channels,
+                        AcquisitionDimension.PROJECTION: self.config.system.num_positions,
+                        AcquisitionDimension.VERTICAL: self.config.panel.num_pixels[1],
+                        AcquisitionDimension.HORIZONTAL: self.config.panel.num_pixels[0]}
 
-        shape_dict = {AcquisitionDimension.CHANNEL: self.config.channels.num_channels,
-                      AcquisitionDimension.ANGLE: self.config.angles.num_positions,
-                      AcquisitionDimension.VERTICAL: self.config.panel.num_pixels[1],
-                      AcquisitionDimension.HORIZONTAL: self.config.panel.num_pixels[0]}
+        
         return tuple(shape_dict[label] for label in self.dimension_labels)
 
     @property
     def dimension_labels(self):
         labels_default = AcquisitionDimension.get_order_for_engine("cil")
 
-        shape_default = [self.config.channels.num_channels,
-                            self.config.angles.num_positions,
-                            self.config.panel.num_pixels[1],
-                            self.config.panel.num_pixels[0]
-                            ]
+        # We are using angles, not per-projection geometry
+        if self.config.system.geometry != "cone_flex":
+            shape_default = [self.config.channels.num_channels,
+                                self.config.angles.num_positions,
```

Comment:
@gfardell suggests that this could be possibly accessed by `self.num_projections` and would cover both cases

Reply:
updated